src/World.h

```diff
@@ -819,6 +822,15 @@ class cWorld :
 		virtual void Execute(void) override;
 	} ;
 
+	struct EntityRemovalQueueItem
+	{
+		EntityRemovalQueueItem(cChunk * a_Chunk, cEntity * a_Entity, bool a_FreeMemory);
+		cChunk * m_Chunk;
+		cEntity * m_Entity;
+		bool m_FreeMemory;
+	};
+
+	std::list<EntityRemovalQueueItem> m_EntityRemovalQueue;
```

Comment:
Oh yeah, mobs are ticked separately (ugh). In that case, move the TickMobs call to after m_ChunkMap->Tick, and now its guaranteed :)